File: packages/ui/db/entity/CoordinateOptional.ts
import { type ICoordinateOptional } from "models";
import { EntitySchema } from "typeorm";

// Embeddable Coordinate columns that are nullable
export type CoordinateOptional = ICoordinateOptional;

const CoordinateOptionalSchema = new EntitySchema<CoordinateOptional>({
  name: "coordinate_optional",
  columns: {
    lat: {
      type: "decimal",
      nullable: true,
    },
    lng: {
      type: "decimal",
      nullable: true,
    },
  },
});

export default CoordinateOptionalSchema;

File: go-server/models/coordinate_optional.go
package models

type CoordinateOptional struct {
	Lat *float64 `json:"lat,omitzero"`
	Lng *float64 `json:"lng,omitzero"`
}

func (c CoordinateOptional) IsZero() bool {
	return c.Lat == nil && c.Lng == nil
}
